src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/repository/OffenderAssessmentRepository.kt

```diff
@@ -1,24 +1,16 @@
 package uk.gov.justice.hmpps.prison.repository.jpa.repository
 
-import org.springframework.data.jpa.repository.Query
+import org.springframework.data.jpa.repository.EntityGraph
+import org.springframework.data.jpa.repository.EntityGraph.EntityGraphType.FETCH
 import org.springframework.data.repository.CrudRepository
 import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenderAssessment
 import java.util.Optional
 
 interface OffenderAssessmentRepository : CrudRepository<OffenderAssessment, OffenderAssessment.Pk> {
   fun findByBookingIdAndAssessmentSeq(bookingId: Long, assessmentSeq: Int): Optional<OffenderAssessment>
 
-  @Query(
-    """
-        SELECT
-          oa FROM OffenderAssessment oa 
-          INNER JOIN oa.offenderBooking booking
-          INNER JOIN booking.offender offender 
-          INNER JOIN oa.assessmentType assessment
-        WHERE
-          offender.nomsId IN (:offenderNos) AND assessment.cellSharingAlertFlag = 'Y'
-        ORDER BY oa.assessmentDate DESC, oa.assessmentSeq DESC
-        """,
-  )
-  fun findByCsraAssessmentAndByOffenderNosOrderByLatestFirst(offenderNos: List<String>): List<OffenderAssessment>
```

Comment:
This is the significant change to this PR.
The native query now uses the derived named query instead and the new query ("findWithDetails...") also uses the same logic.